check_metadata.py

```diff
@@ -72,19 +75,35 @@ def checkFile(filePattern):
     print("")
 
 
-def verifyNoBlacklistWords(fileContents, filelocation):
+def verifyNoDenyListWords(fileContents, filelocation):
     fileContents = fileContents.split('/')
     for word in fileContents:
-        if word in blacklist:
+        if word in denyList:
             print("ERROR -- Found in " + filelocation)
-            sys.exit("ERROR -- " + word + " found, and is on the blacklist.")
-
-def checkStringLength (word):
-    length = len(word)
-    if  length == 40 or length == 20:
-        if warn:
-            return "WARNING -- " + word + " is " + str(length) + " characters long"
+            sys.exit("ERROR -- " + word + " found, and is not allowed.")
 
+def characterScan(fileContents, filelocation):
+    twenty = re.findall("[^A-Z0-9][A][ACGIKNPRS][A-Z]{2}[A-Z0-9]{16}[^A-Z0-9]", fileContents)
+    if (twenty) :
+        for word in twenty:
+            if word[1:-1] in allowList:
+                continue
+            print("ERROR -- " + word[1:-1] + " Found in " + filelocation)
+            # sys.exit("ERROR -- 20 character string found, and might be a secret access key.") 
```

Comment:
Are you sure this line should be commented out?

Reply:
Great catch :)

I was trying to get a list of all patterns that matched to add to the allow list.